modules/common/src/main/String.scala

```diff
@@ -21,18 +21,28 @@ object String:
     try play.utils.UriEncoding.decodePath(input, "UTF-8").some
     catch case _: play.utils.InvalidUriEncodingException => None
 
-  def isShouting(text: String) =
+  def isShouting(text: String): Boolean =
     text.lengthIs >= 5 && {
       import java.lang.Character.*
       // true if >1/2 of the latin letters are uppercase
-      text.take(80).replace("O-O", "o-o").foldLeft(0) { (i, c) =>
-        getType(c) match
-          case UPPERCASE_LETTER => i + 1
-          case LOWERCASE_LETTER => i - 1
-          case _                => i
-      } > 0
+      if text.contains('\ue666') then
```

Comment:
No. The broadcast encoding logic must not spill over unrelated section of the code, such as this one.

Reply:
What is your suggested way? Do I add 128 to ascii value of each character of chapterId in `encodeMsg` and then get back the chapterId by subtracting 128? So the chapterId doesn't affect the shouting mechanism.

Reply:
```js
let encodedChapterId = '';
for (let i = 0; i < chapterId.length; i++) {
  encodedChapterId += String.fromCharCode(chapterId.charCodeAt(i) + 128);
}
chapterId = encodedChapterId;
```
Something like this?

Reply:
`isShouting` should only be applied to the text that was actually typed by the user.